kernel/core/datamanagement/memory.cpp

```diff
@@ -3748,94 +3748,162 @@ std::vector<mu::value_type> Memory::getIndex(size_t col, const std::vector<mu::v
     return vIndex;
 }
 
-
 /////////////////////////////////////////////////
 /// \brief Calculates the simples form of a ANOVA
 /// F test
 ///
-/// \param colCategories size_t
+/// \param colCategories VectorIndex&
 /// \param colValues size_t
 /// \param _vIndex const VectorIndex&
 /// \param significance double
 /// \return AnovaResult
 ///
 /////////////////////////////////////////////////
-AnovaResult Memory::getOneWayAnova(size_t colCategories, size_t colValues, const VectorIndex& _vIndex, double significance) const
+std::vector<AnovaResult> Memory::getAnova(const VectorIndex& colCategories, size_t colValues, const VectorIndex& _vIndex, double significance) const
 {
     // Get indices
-    if (colCategories > memArray.size()
-        || !memArray[colCategories]
-        || memArray[colCategories]->m_type != TableColumn::TYPE_CATEGORICAL
+    if (colCategories[0] > memArray.size()
+        || !memArray[colCategories[0]]
+        || memArray[colCategories[0]]->m_type != TableColumn::TYPE_CATEGORICAL
         || significance >= 1.0
-        || significance <= 0.0)
+        || significance <= 0.0
+        || colCategories.size() > 2)
     {
         AnovaResult res;
         res.m_FRatio = NAN;
-        return res;
+        return std::vector<AnovaResult> {res};
     }
 
-    _vIndex.setOpenEndIndex(getElemsInColumn(colCategories)-1);
-    Memory _mem(2);
-    _mem.memArray[0].reset(memArray[colCategories]->copy(_vIndex));
-    _mem.memArray[1].reset(memArray[colValues]->copy(_vIndex));
+    bool isTwoWay = colCategories.size() == 2;
+    _vIndex.setOpenEndIndex(getElemsInColumn(colCategories[0])-1);

```

Comment:
Actually, they should all be of the same size, since every value has to be in one of the categories. 
But that means we should add a check at the beginning if all columns are of the same size

Reply:
You can check or you can simply use the shortest. Both approaches are valid.